### PR TITLE
Use Result in API to more closely match Tsdl

### LINKED ISSUE #5: Use Result in API
We should use `Result` where appropriate in order to more closely match Tsdl's API, and to give more convenient access to error messages.  Target Tsdl 0.9.\* while we're at it.

### PR DESCRIPTION
**Pull request by dagoof**
This should resolve #5

### CODE REVIEW

File: src/tsdl_image.ml [hunk 1]
```diff
@@ -1,6 +1,11 @@
 open Ctypes
 open Foreign
 open Tsdl
+open Result
+
+type 'a result = 'a Sdl.result
```
Comment (tokenrove) [inline src/tsdl_image.ml:6]: I think this type and `error` should go inside `module Image` for consistency with tsdl.

**Review by tokenrove — COMMENTED** [review]
Just a minor update -- I started the work to merge this and your other PR, which has forced me to fix the broken CI builds and a bunch of other little things.  I can make the change I've suggested here when I do the merge (I'm also going to make some minor adjustments in the tsdl-ttf PR), but I wanted to let you know that despite the delay, this is slowly moving forward.  (Wrangling with Travis CI is always a hassle.)

File: src/tsdl_image.ml (hunk 1 above)
Comment (dagoof) [reply src/tsdl_image.ml:6]: agreed -- happy to commit it in if it helps; but don't want to get in your way at this point

**Comment by tokenrove** [comment]
Merged as commit 9bf2918685f70896a2467838b69b88856ef32ee8 rather than through this PR due to minor adjustments.  Thanks for your contribution and your patience.

0.2.0 should be available once [the OPAM PR](https://github.com/ocaml/opam-repository/pull/8270) clears.  tsdl-ttf and tsdl-mixer should follow shortly.

**Pull request closed without merging**